filters/polarity/filter.py

```diff
@@ -0,0 +1,25 @@
+from interfaces.SentencePairOperation import SentencePairOperation
+from tasks.TaskTypes import TaskType
+from initialize import spacy_nlp
+from spacytextblob.spacytextblob import SpacyTextBlob
+import spacy
+
+
+class TextRetainsPolarity(SentencePairOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self, diff_allowed: float = 0.5):
+        super().__init__()
+        self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+        if "spacytextblob" not in self.nlp.pipe_names:
+            self.nlp.add_pipe("spacytextblob")
+        self.diff_allowed = diff_allowed
+
+    def filter(self, sentence1: str = None, sentence2: str = None) -> bool:
+        text1 = self.nlp(sentence1, disable=["parser", "tagger", "ner", 'lemmatizer'])
+        text2 = self.nlp(sentence2, disable=["parser", "tagger", "ner", 'lemmatizer'])
+        if abs(text1._.polarity - text2._.polarity) <= self.diff_allowed:
```

Comment:
I wonder if you have considered cases on the edge: e.g., a sentence is marginally positive, and after the change it becomes marginally negative --- would there be cases where the label is changed but the diff still satisfy the `self.diff_allowed ` constraint? Should there be a global threshold?

Reply:
The outlined case is possible but in terms of sentiment analysis tasks, I would say it is acceptable. Considering that sentiment scoring tools are not always consistent and cases of near-neutral polarity are mainly problematic for polarity interpretation, we don't want to restrict transformed cases here much. However, potentially this could be a nice argument to introduce: to allow or not to allow any kind of sentiment shift (even from -0.1 to 0.1).

Reply:
I added another argument that can do a stricter polarity check.